src/CloudEvent.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+class CloudEvent
+{
+    // Required Fields
+    public $id;
+    public $source;
+    public $specversion;
+    public $type;
+    
+    // Optional Fields
+    public $datacontenttype;
+    public $dataschema;
+    public $subject;
+    public $time;
+    public $data;
```

Comment:
It's a bit of a pain but I'd rather have these be private and use getters. that way we can preserve strong typing.

Reply:
Fixed.

Reply:
I realize now this is the pattern we are using for `Context`, so for consistency we can keep this as-is!